### DISCUSSION TITLE
JSX-like syntax extension

### DISCUSSION THREAD

**Discussion by pabloruizetc** [reflex-dev/reflex#1582] [Ideas]
This project is amazing! I wonder if you've thought about implementing a syntax extension like JSX does with JavaScript? A syntactic sugar for Python that allows you to write HTML-like markup inside a Python file. Something similar has been developed before (.[pyxl](https://github.com/gvanrossum/pyxl3)). Only the .pyxl code would need to be transpiled to .py at build time. I think that it is totally in line with the project and that it can greatly improve the developer experience.

**Comment by Alek99** [comment]
This is something we have thought about, but by not doing everything in Python, you miss out on some flexibility and end up recreating a different language of sorts. 

Not to say this is something we would never do, I think we would need to evaluate how exactly this would look.
